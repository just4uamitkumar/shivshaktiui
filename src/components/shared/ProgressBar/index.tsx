import { useEffect, useState } from "react";
import { MAX, MIN } from "./constants";
import './styles.scss'
import { Stack } from "@mui/material";


interface Props{
    value:number;
    onComplete:() => void;
}

const ProgressBar:React.FC<Props> = ({ value = 0, onComplete = () => {} }) => {
  const [percent, setPercent] = useState<number>(value);

  useEffect(() => {
    setPercent(Math.min(Math.max(value, MIN), MAX));

    if (value >= MAX) {
      onComplete();
    }
  }, [value]);

  return (
    <Stack className="progress">
      <span
        style={{
          color: percent > 49 ? "white" : "black"
        }}
      >
        {percent.toFixed()}%
      </span>
      <Stack
        // style={{ width: `${percent}%` }}
        style={{
          transform: `scaleX(${percent / MAX})`,
          transformOrigin: "left"
        }}
        aria-valuemin={MIN}
        aria-valuemax={MAX}
        aria-valuenow={percent}
        role="progressbar"
      />
    </Stack>
  );
}

export default ProgressBar