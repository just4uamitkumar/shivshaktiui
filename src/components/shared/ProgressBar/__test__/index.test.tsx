import { describe, it, expect, vi } from "vitest";
import { render, screen } from "@testing-library/react";
import ProgressBar from "../index";

describe("ProgressBar Component", () => {
  it("matches snapshot", () => {
    const { container } = render(
      <ProgressBar value={50} onComplete={() => {}} />
    );
    expect(container).toMatchSnapshot();
  });
  it("renders with correct initial value", () => {
    render(<ProgressBar value={50} onComplete={() => {}} />);
    const progressText = screen.getByText("50%");
    expect(progressText).toBeInTheDocument();
  });
  it("updates value correctly", () => {
    const onCompleteMock = vi.fn();
    const { rerender } = render(
      <ProgressBar value={50} onComplete={onCompleteMock} />
    );
    expect(screen.getByText("50%")).toBeInTheDocument();

    rerender(<ProgressBar value={75} onComplete={onCompleteMock} />);
    expect(screen.getByText("75%")).toBeInTheDocument();
  });
  it("calls onComplete when value reaches MAX", () => {
    const onCompleteMock = vi.fn();
    render(<ProgressBar value={100} onComplete={onCompleteMock} />);
    expect(onCompleteMock).toHaveBeenCalled();
  });
});
