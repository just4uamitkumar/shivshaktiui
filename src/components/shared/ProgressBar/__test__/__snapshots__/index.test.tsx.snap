// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProgressBar Component > matches snapshot 1`] = `
<div>
  <div
    class="MuiStack-root progress css-nen11g-MuiStack-root"
  >
    <span
      style="color: white;"
    >
      50
      %
    </span>
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="50"
      class="MuiStack-root css-nen11g-MuiStack-root"
      role="progressbar"
      style="transform: scaleX(0.5); transform-origin: left;"
    />
  </div>
</div>
`;
